app/models/game.rb

```diff
@@ -14,6 +14,30 @@ def populate!
     populate_right_white_half!
   end
 
+  def white_check?
+    white_king = pieces.find_by(type: 'King', color: 'white')
+    pieces.each do |piece|
```

Comment:
Loop all the pieces? Couldn't we save some horsepower by first filtering on opposite color?